src/shaders/symbol_icon.vertex.glsl

```diff
@@ -84,10 +91,14 @@ void main() {
         symbol_rotation = atan((b.y - a.y) / u_aspect_ratio, b.x - a.x);
     }
 
+    vec3 proj_pos = mix_globe_mercator(
+        u_inv_rot_matrix, tileAnchor, vec3(a_projected_pos.xy, anchorZ),
+        u_tile_id, u_merc_center, u_zoom_transition);
+
 #ifdef PROJECTED_POS_ON_VIEWPORT
```

Comment:
the `PROJECTED_POS_ON_VIEWPORT` define is used to separate coordinates that are in normalized ecef space (point labels) from coordinates in label spaces (viewport or tile). Elevation can be added only to point placed symbols as they exists in the correct space. Name of the define is a bit misleading and outdated as it's mostly about point placement vs line placement.

I don't think this will work with line labels as you're potentially interpolating between viewport coordinates and map coordinates.

![line_label_transition](https://user-images.githubusercontent.com/55925868/132740791-dac55432-c434-46d5-80b5-70c0d3c8d8a5.gif)

Do you think if we should skip those for now or find a solution?

Reply:
Good point. Thinking about this, a potential way forward about this could be to leverage the fact that we have runtime updates of collision circle placement, so we could readjust their position by encoding the transition interpolation in `projection.projectTilePoint` (only for line labels). Investigating that a bit.

Reply:
I have temporarily disabled transition of line place labels in https://github.com/mapbox/mapbox-gl-js/pull/10999/commits/40beb78d9714be2ad8a2c2e9c35ee2b709e01f3c to have something we can use for now, and will continue investigation of the approach I mentioned above as I think that could be something to pursue further.